src/app/pages/home/containers/home-page.component.ts

```diff
@@ -277,13 +276,19 @@ export class HomePage {
         introduction: assessment.startText[language],
         endText: assessment.endText[language],
         questions: assessment.questions,
-        associatedTask: startQuestionnaireTask
+        associatedTask: startQuestionnaireTask,
+        assessment: assessment,
+        isLastTask: false
       }
-      this.controller.updateAssessmentIntroduction(assessment)
+
+      this.tasksService.isLastTask(startQuestionnaireTask).then(lastTask => {
+        if (lastTask) params.isLastTask = true
+      })
```

Comment:
This should be calculated before the next statement (navigate to next page) is executed. But from what I can see this will be run in the background (async) and we may lose this info if navigated to different page before this is processed. Please feel free to correct me.

Reply:
Yup this is fixed in #172